language/tools/vm-genesis/src/lib.rs

```diff
@@ -543,103 +539,114 @@ fn verify_genesis_write_set(events: &[ContractEvent]) {
 
 /// Generate an artificial genesis `ChangeSet` for testing
 pub fn generate_genesis_change_set_for_testing(stdlib_options: StdLibOptions) -> ChangeSet {
-    let stdlib_modules = stdlib_modules(stdlib_options);
-    let swarm = libra_config::generator::validator_swarm_for_testing(10);
-
-    encode_genesis_change_set(
-        &GENESIS_KEYPAIR.1,
-        &GENESIS_KEYPAIR.1,
-        &operator_assignments(&swarm.nodes),
-        &operator_registrations(&swarm.nodes),
-        stdlib_modules,
+    generate_test_genesis(
+        &stdlib_modules(stdlib_options),
         VMPublishingOption::open(),
-        ChainId::test(),
+        None,
     )
     .0
 }
 
 /// Generate an artificial genesis `ChangeSet` for testing
 pub fn generate_genesis_type_mapping() -> BTreeMap<Vec<u8>, StructTag> {
-    let stdlib_modules = stdlib_modules(StdLibOptions::Compiled);
-    let swarm = libra_config::generator::validator_swarm_for_testing(10);
-
-    encode_genesis_change_set(
-        &GENESIS_KEYPAIR.1,
-        &GENESIS_KEYPAIR.1,
-        &operator_assignments(&swarm.nodes),
-        &operator_registrations(&swarm.nodes),
-        stdlib_modules,
+    generate_test_genesis(
+        &stdlib_modules(StdLibOptions::Compiled),
         VMPublishingOption::open(),
-        ChainId::test(),
+        None,
     )
     .1
 }
 
-/// Generates an artificial set of OperatorAssignments using the given node configurations for
-/// testing.
-pub fn operator_assignments(node_configs: &[NodeConfig]) -> Vec<OperatorAssignment> {
-    node_configs
-        .iter()
-        .enumerate()
-        .map(|(idx, n)| {
-            let test_config = n.test.as_ref().unwrap();
-            let owner_key = test_config.owner_key.as_ref().unwrap().public_key();
-            let operator_key = test_config.operator_key.as_ref().unwrap().public_key();
-            let operator_account = account_address::from_public_key(&operator_key);
-            let name = idx.to_string().as_bytes().to_vec();
-            let set_operator_script = transaction_builder::encode_set_validator_operator_script(
-                name.clone(),
-                operator_account,
-            );
+pub fn test_genesis_transaction() -> Transaction {
```

Comment:
why those functions are not with cfg(test)?

Reply:
they are used throughout the codebase, we'd have to add flags all over the place for it. I'm slowly getting to the point where I think having test utils exposed isn't that bad. Cargo doesn't play very kindly with these type of features and we aren't pulling in additional crates.